detox/src/devices/drivers/android/genycloud/GenyCloudDriver.js

```diff
@@ -17,43 +17,48 @@ class GenyCloudDriver extends AndroidDriver {
 
     const exec = new GenyCloudExec();
     const instanceNaming = new InstanceNaming(); // TODO should consider a permissive impl for debug/dev mode. Maybe even a custom arg in package.json (Detox > ... > genycloud > sharedAccount: false)
-    this.recipeService = new RecipesService(exec, logger);
-    this.instanceLookupService = new InstanceLookupService(exec, instanceNaming, this.deviceRegistry);
-    this.instanceLifecycleService = new InstanceLifecycleService(exec, instanceNaming);
-    this.deviceQueryHelper = new DeviceQueryHelper(this.recipeService);
-    this.allocationHelper = new AllocationHelper(this.instanceLookupService, this.instanceLifecycleService);
+    this._deviceRegistry = DeviceRegistry.forAndroid();
+
+    const recipeService = new RecipesService(exec, logger);
+    const instanceLookupService = new InstanceLookupService(exec, instanceNaming, this._deviceRegistry);
+    const instanceLifecycleService = new InstanceLifecycleService(exec, instanceNaming);
+    this._deviceQueryHelper = new DeviceQueryHelper(recipeService);
+    this._deviceAllocator = new GenyCloudDeviceAllocator(this._deviceRegistry, instanceLookupService, instanceLifecycleService);
   }
 
   get name() {
     return this._name;
   }
 
   async acquireFreeDevice(deviceQuery) {
-    const recipe = await this.deviceQueryHelper.getRecipeFromQuery(deviceQuery);
-    this.assertRecipe(deviceQuery, recipe);
+    const recipe = await this._deviceQueryHelper.getRecipeFromQuery(deviceQuery);
+    this._assertRecipe(deviceQuery, recipe);
 
-    const cookie = { coldBoot: false };
-    const adbName = await this.allocateDevice({
-      recipe,
-      cookie,
-      toString: () => recipe.toString(),
-    });
+    const { instance, isNew } = await this._deviceAllocator.allocateDevice(recipe);
+    const { adbName, uuid } = instance;
```

Comment:
is `uuid` = `recipeUUID`? I'd name it so, if this is the case, otherwise I'm not sure what it means